planet/sync/features.py

```diff
@@ -0,0 +1,200 @@
+# Copyright 2025 Planet Labs PBC.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+from typing import Iterator, Optional, Union
+from planet.clients.features import FeaturesClient
+from planet.http import Session
+from planet.models import Feature, GeoInterface
+
+
+class FeaturesAPI:
+
+    _client: FeaturesClient
+
+    def __init__(self, session: Session, base_url: Optional[str] = None):
+        """
+        Parameters:
+            session: Open session connected to server.
+            base_url: The base URL to use. Defaults to production Features API
+                base url.
+        """
+        self._client = FeaturesClient(session, base_url)
+
+    def list_collections(self) -> Iterator[dict]:
+        """
+        List the feature collections you have access to.
+
+        Example:
+
+        ```
+        pl = Planet()
+        collections = pl.features.list_collections()
+        for collection in collections:
+            print(collection)
+        ```
+        """
+        collections = self._client.list_collections()
+
+        try:
+            while True:
+                yield self._client._call_sync(collections.__anext__())
+        except StopAsyncIteration:
+            pass
+
+    def get_collection(self, collection_id: str) -> dict:
+        """
+        Get collection metadata.
+
+        note: when looking up or adding features to a collection,
+        you can pass the collection ID directly to the add_features
+        or list_features methods. It is not necessary to look up
+        the collection metadata first.
+        """
+        collection = self._client.get_collection(collection_id)
+        return self._client._call_sync(collection)
+
+    def create_collection(self,
+                          title: str,
+                          description: Optional[str] = None) -> str:
+        """
+        Create a new collection with the given title and description, returning the collection id.
+
+        Example:
+
+        ```
+        pl = Planet()
+        collection_id = pl.features.create_collection(
+          title="My Collection",
+          description="a test collection"
+        )
+        ```
+        """
+        collection = self._client.create_collection(title, description)
+        return self._client._call_sync(collection)
+
+    def list_items(self, collection_id: str) -> Iterator[Feature]:
```

Comment:
I was using the sync features client today, and discovered this needs to expose `limit`.
The underlying list FeaturesClient defaults to `limit=10`, so it is currently impossible to list more than 10 items using the sync client

I presume list_collections is similar

Reply:
added in 70e411b